tests/dm_test.py

```diff
@@ -39,6 +39,21 @@ def _clean_up(self):
 
         os.unlink(self.dev_file)
 
+class DevMapperGetSubsystemFromName(DevMapperTestCase):
+    def _destroy_lvm(self):
+        run("vgremove --yes libbd_dm_tests >/dev/null 2>&1")
+        run("pvremove --yes %s >/dev/null 2>&1" % self.loop_dev)
+
+    def test_get_subsystem_from_name(self):
+        """Verify that it is possible to get device subsystem from its name"""
+        run("pvcreate -ff -y %s >/dev/null 2>&1" % self.loop_dev)
+        run("vgcreate -s10M libbd_dm_tests %s >/dev/null 2>&1" % self.loop_dev)
+        run("lvcreate -n subsystem_test -L50M libbd_dm_tests >/dev/null 2>&1")
+        self.addCleanup(self._destroy_lvm)
+
+        subsystem = BlockDev.dm_get_subsystem_from_name("libbd_dm_tests-subsystem_test")
+        self.assertEqual(subsystem, "LVM")
```

Comment:
Would be great to also have a test for some other subsystem. The other easiest one is probably LUKS (crypt).

Reply:
Fixed. I have written test for luks device. Since it requires reinitialization of the resources used in LVM test, I have put it into a new function.